### PR TITLE
Localize FXIOS-4800 [v106] Remove obsolete strings from AppMenu

### PR DESCRIPTION
**Pull request by flodolo**
[FXIOS-4800](https://mozilla-hub.atlassian.net/browse/FXIOS-4800) https://github.com/mozilla-mobile/firefox-ios/issues/11665

### CODE REVIEW

File: Client/Frontend/Strings.swift [hunk 1]
```diff
@@ -9,7 +9,6 @@ import Foundation
  Strings with the following IDs may be removed after confirming with Product:
  - `ActivityStream.Pocket.SectionTitle`      --> replaced by `FirefoxHome.Pocket.SectionTitle`
  - `ActivityStream.Pocket.SectionTitle2`    --> replaced by `FirefoxHome.Pocket.SectionTitle`
```
Comment (flodolo) [inline Client/Frontend/Strings.swift:11]: These two strings are actually unused. Who can sign off on removing them?

File: Client/Frontend/Strings.swift [hunk 2]
```diff
@@ -9,7 +9,6 @@ import Foundation
...
- - All Strings marked under "unused" in AppMenu struct
  - "HistoryPanel.ClearHistoryMenuTitle" string
```
Comment (flodolo) [inline Client/Frontend/Strings.swift:12]: This seems still in use?

Definition: https://github.com/mozilla-mobile/firefox-ios/blob/main/Client/Frontend/Strings.swift#L1691-L1695
Use: https://github.com/mozilla-mobile/firefox-ios/blob/main/Client/Frontend/Library/ClearHistorySheetProvider.swift#L34

Comment (flodolo) [reply Client/Frontend/Strings.swift:12]: Gosh, this is so confusing.
* There is a `LibraryPanel.History.ClearHistoryMenuTitle` (string ID: `LibraryPanel.History.ClearHistoryMenuTitle.v100`). 
* And there's a `.ClearHistoryMenuTitle` (string ID: `HistoryPanel.ClearHistoryMenuTitle`), which is unused.

**Comment by lmarceau** [comment]
Build is green [here](https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/a5f32e6b-9543-4215-bd38-22ab8ad9b3f1)
Screen Shot 2022-08-23 at 11 13 04 AM

**Comment by flodolo** [comment]
Should I be worried about the `testTopSitesRemoveAllExceptDefaultClearPrivateData` failure?
https://github.com/mozilla-mobile/firefox-ios/pull/11676/checks?check_run_id=7976333222

I see other PRs don't have that, and I'm not sure if this is caused by this change
https://github.com/mozilla-mobile/firefox-ios/compare/3ba6176bb06901482fb70b9abe9c180a9bd9e1ac..d39ef28e0b6359103a4ce756bf342e796fb4e163

File: Client/Frontend/Strings.swift (hunk 1 above)
Comment (lmarceau) [reply Client/Frontend/Strings.swift:11]: @dnarcese 👀
Those strings refers to unused strings:
```
    public static let ASPocketTitle = MZLocalizedString(
        "ActivityStream.Pocket.SectionTitle",
        value: "Trending on Pocket",
        comment: "Section title label for Recommended by Pocket section",
        lastUpdated: .unknown)
    public static let ASPocketTitle2 = MZLocalizedString(
        "ActivityStream.Pocket.SectionTitle2",
        value: "Recommended by Pocket",
        comment: "Section title label for Recommended by Pocket section",
        lastUpdated: .unknown)
```

Comment (flodolo) [reply Client/Frontend/Strings.swift:11]: Correct, what's unclear is why the note about product manager. 

Should I just remove them here?

Comment (lmarceau) [reply Client/Frontend/Strings.swift:11]: We wanna confirm with Product to make sure there's no plan to reuse those strings in the near future. Hence why confirming with @dnarcese is recommended

Comment (flodolo) [reply Client/Frontend/Strings.swift:11]: Sounds good to me. Probably OK to merge this one then, and remove them afterwards? There are still a lot to remove… (need to update the list).

**Comment by lmarceau** [comment]
I'll merge, and let's open another PR for ASPocketTitle and ASPocketTitle2 if they need to be removed as well.

File: Client/Frontend/Strings.swift (hunk 1 above)
Comment (dnarcese) [reply Client/Frontend/Strings.swift:11]: They can be removed.

**Pull request merged**